src/scripts/upload.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
```

Comment:
let's also not do a `set -e`. You are already handling the invalid input. In case the curl request fails, we should not fail the test suite for the user, since it's not their fault.